bridge/setu/processor/checkpoint.go

```diff
@@ -528,6 +533,29 @@ func (cp *CheckpointProcessor) nextExpectedCheckpoint(checkpointContext *Checkpo
 	}), nil
 }
 
+func (cp *CheckpointProcessor) checkCrossChain(start uint64, end uint64, rootChain string,
+	maxCheckpointLengthParam uint64,
+) bool {
+	isOpen, rootChainIsOpen, maxLength := cp.getDynamicCheckpointProposal(rootChain)
+
+	if !isOpen || !rootChainIsOpen {
+		return true
+	}
+
+	if maxCheckpointLengthParam < uint64(maxLength) {
+		cp.Logger.Error("proposal feature-dynamic-checkpoint maxlength is too long",
+			"maxLength", maxLength, "MaxCheckpointLength", maxCheckpointLengthParam)
+
+		return true
```

Comment:
What if the end - start + 1 < maxCheckpointLengthParam but maxCheckpointLengthParam < maxLength ?

I wonder should the case always be true?

Reply:
in case :checkpoint length<=maxCheckpointLengthParam<maxLength
There will no checkpoint to submit when no cross chain tx in chain.